src/block/block_cache.c

```diff
@@ -45,8 +70,6 @@ __blkcache_alloc(WT_SESSION_IMPL *session, size_t size, void **retp)
     else if (blkcache->type == BLKCACHE_NVRAM) {
 #ifdef HAVE_LIBMEMKIND
         *retp = memkind_malloc(blkcache->pmem_kind, size);
-        if (*retp == NULL)
```

Comment:
Because `__wt_malloc()` can return any error, including `WT_PANIC`, meaning "your database has been corrupted, shut down now". Collapsing all errors from `__wt_malloc()` to mean "no memory available" would be a serious bug.